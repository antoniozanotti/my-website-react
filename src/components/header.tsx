import { TzButton, TzToggleTheme } from "topaz-react";
import SocialMedia from "@components/social-media";
import Terminal from "@components/terminal";
import Heading from "@components/heading";

export default function Header() {
  return (
    <header className="h-screen flex flex-col justify-between">
      <div className="pt-4 sm:pt-8 lg:pt-16 flex justify-end">
        <TzToggleTheme />
      </div>
      <div>
        <Heading level="h1" title="AntÃ´nio Zanotti" />
        <Terminal className="mt-[38px]" />
      </div>
      <SocialMedia />
      <div className="pb-4 sm:pb-8 lg:pb-16 flex place-content-center">
        <TzButton
          iconName="ArrowDownIcon"
          variant="secondary"
          onClick={() => {
            window.scroll({
              top: window.innerHeight,
              left: 0,
              behavior: "smooth",
            });
          }}
        />
      </div>
    </header>
  );
}
