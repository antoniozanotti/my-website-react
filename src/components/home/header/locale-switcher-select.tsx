"use client";

import { useRouter } from "next/navigation";
import { ChangeEvent, ReactNode, useTransition } from "react";

type Props = {
  children: ReactNode;
  defaultValue: string;
  label: string;
};

export function LocaleSwitcherSelect({ children, defaultValue, label }: Props) {
  const router = useRouter();
  const [isPending, startTransition] = useTransition();

  function onSelectChange(event: ChangeEvent<HTMLSelectElement>) {
    const nextLocale = event.target.value;
    startTransition(() => {
      router.push("/"+nextLocale);
    });
  }

  let labelClasses = "relative relative flex justify-around";
  if (isPending) {
    labelClasses += " transition-opacity [&:disabled]:opacity-30";
  }

  return (
    <label className={labelClasses}>
      <p className="sr-only">{label}</p>
      <select
        className="inline-flex appearance-none bg-secondary dark:bg-dark-secondary w-[153px] lg:w-[173px] h-[38px] lg:h-[48px] rounded-full cursor-pointer py-0 pl-4 pr-6"
        defaultValue={defaultValue}
        disabled={isPending}
        onChange={onSelectChange}
      >
        {children}
      </select>
      <span className="pointer-events-none absolute right-4 top-[6px] lg:top-[8px]">
        âŒ„
      </span>
    </label>
  );
}
