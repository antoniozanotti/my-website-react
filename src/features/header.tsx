import { TzButton, TzToggleTheme } from "topaz-react";
import { SocialMedia } from "@features/social-media";
import { Terminal } from "@features/terminal";
import { Heading } from "@features/heading";

export const Header = () => {
  return (
    <header className="h-screen">
      <TzToggleTheme />
      <div className="">
        <Heading level="h1" title="AntÃ´nio Zanotti" className="" />
        <Terminal className="" />
        <SocialMedia className="" />
      </div>
      <TzButton iconName="ArrowDownIcon" variant="secondary" className="animate-bounce" />
    </header>
  );
};
