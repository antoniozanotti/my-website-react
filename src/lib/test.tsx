import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
