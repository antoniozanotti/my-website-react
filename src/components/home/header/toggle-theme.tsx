'use client';
import React from 'react';
import { useAppSelector, useAppDispatch, useAppStore } from '@/lib/hooks';
import { toggleTheme } from '@/lib/features/themeSlice';
import { TzToggleTheme } from "topaz-react";

export function ToggleTheme() {
  const store = useAppStore();
  const theme = useAppSelector((state) => state.theme.value);
  const dispatch = useAppDispatch();

  const handleClick = () => {
    dispatch(toggleTheme());
  };

  return (
    <TzToggleTheme onClick={handleClick} />
  );
}
