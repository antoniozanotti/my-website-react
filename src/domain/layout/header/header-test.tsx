import { render } from "@/lib/test";
import { Header } from "./header";

describe("Header", () => {
  test("should render header", async () => {
    const { container } = render(<Header />);
    const header = container.querySelector("header");

    expect(header).toBeVisible();
  });
});
