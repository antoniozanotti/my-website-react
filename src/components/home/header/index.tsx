"use client";
import { TzButton } from "topaz-react";
import { ToggleTheme } from "./toggle-theme";
import SocialMedia from "@/components/common/social-media";
import Terminal from "./terminal";
import Heading from "@/components/ui/heading";

export default function Index() {
  return (
    <header className="h-[100vh]">
      <div className="h-[100svh] flex flex-col justify-between">
        <div className="pt-4 sm:pt-8 lg:pt-16 flex justify-end">
          <ToggleTheme />
        </div>
        <div>
          <Heading level="h1" title="AntÃ´nio Zanotti" />
          <Terminal className="mt-[38px]" />
        </div>
        <SocialMedia />
        <div className="pb-4 sm:pb-8 lg:pb-16 flex place-content-center">
          <TzButton
            iconName="ArrowDownIcon"
            variant="secondary"
            onClick={() => {
              window.scroll({
                top: document.getElementById("about")?.offsetTop,
                left: 0,
                behavior: "smooth",
              });
            }}
          />
        </div>
      </div>
    </header>
  );
}
