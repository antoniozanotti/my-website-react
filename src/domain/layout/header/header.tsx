"use client";
import { ToggleTheme } from "./toggle-theme/toggle-theme";
import { LocaleSwitcher } from "./locale-switcher/locale-switcher";
import { Heading } from "@/domain/ui/heading/heading";
import { Terminal } from "./terminal/terminal";
import { SocialMedia } from "@/domain/home/social-media/social-media";
import { TzButton } from "topaz-react";

export function Header() {
  return (
    <header className="h-[100vh]">
      <div className="h-[100svh] flex flex-col justify-between">
        <div className="pt-4 sm:pt-8 lg:pt-16 flex justify-end space-x-4">
          <ToggleTheme />
          <LocaleSwitcher />
        </div>
        <div>
          <Heading level="h1" title="AntÃ´nio Zanotti" />
          <Terminal className="mt-[38px]" />
        </div>
        <SocialMedia />
        <div className="pb-4 sm:pb-8 lg:pb-16 flex place-content-center">
          <TzButton
            aria-label="Go to About Section"
            iconName="ArrowDownIcon"
            variant="secondary"
            onClick={useGoToAboutSection}
          />
        </div>
      </div>
    </header>
  );
}

function useGoToAboutSection() {
  window.scroll({
    top: document.getElementById("about")?.offsetTop,
    left: 0,
    behavior: "smooth",
  });
}