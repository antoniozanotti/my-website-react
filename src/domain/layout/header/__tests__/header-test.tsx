import { render } from "@/lib/test";
import { Header } from "../header";

describe("Header", () => {
  it("should render header", async () => {
    const { container } = render(<Header />);
    const header = container.querySelector("header");

    expect(header).toBeVisible();
  });
});
