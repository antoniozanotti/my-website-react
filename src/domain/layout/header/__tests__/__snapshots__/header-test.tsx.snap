// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <header
          class="h-[100vh]"
        >
          <div
            class="h-[100svh] flex flex-col justify-between"
          >
            <div
              class="pt-4 sm:pt-8 lg:pt-16 flex justify-end space-x-4"
            >
              <button
                aria-label="Toggle Theme"
                class="rounded-full cursor-pointer relative flex justify-around bg-secondary dark:bg-dark-secondary w-[104px] h-[38px] lg:w-[132px] lg:h-[48px] "
              >
                <span
                  class="block rounded-full absolute transform duration-300 bg-secondary-2 dark:bg-dark-secondary-2 w-[60px] h-[32px] lg:w-[77px] lg:h-[42px] self-center left-[40px] dark:left-[3px] lg:left-[52px]"
                />
                <svg
                  aria-hidden="true"
                  class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-primary dark:text-dark-primary place-self-center"
                  data-slot="icon"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    MoonIcon
                  </title>
                  <path
                    d="M21.752 15.002A9.72 9.72 0 0 1 18 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 0 0 3 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 0 0 9.002-5.998Z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-primary dark:text-dark-primary place-self-center"
                  data-slot="icon"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    SunIcon
                  </title>
                  <path
                    d="M12 3v2.25m6.364.386-1.591 1.591M21 12h-2.25m-.386 6.364-1.591-1.591M12 18.75V21m-4.773-4.227-1.591 1.591M5.25 12H3m4.227-4.773L5.636 5.636M15.75 12a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0Z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
              <label
                class="relative relative flex justify-around "
              >
                <p
                  class="sr-only"
                >
                  Change language
                </p>
                <select
                  class="inline-flex appearance-none bg-secondary dark:bg-dark-secondary w-[153px] lg:w-[173px] h-[38px] lg:h-[48px] rounded-full cursor-pointer py-0 pl-4 pr-6"
                >
                  <option
                    selected=""
                    value="en"
                  >
                    ðŸ‡ºðŸ‡¸ English
                  </option>
                  <option
                    value="pt"
                  >
                    ðŸ‡§ðŸ‡· PortuguÃªs
                  </option>
                </select>
                <span
                  class="pointer-events-none absolute right-4 top-[6px] lg:top-[8px]"
                >
                  âŒ„
                </span>
              </label>
            </div>
            <div>
              <h1
                class="leading-none text-c9 sm:text-c10 lg:text-c11 text-accent dark:text-dark-accent drop-shadow  "
                id=""
              >
                AntÃ´nio Zanotti
              </h1>
              <p
                class="text-c4 sm:text-c3 lg:text-c5 flex items-center leading-3 lg:leading-5 mt-[38px]"
              >
                <span
                  class="w-2 lg:w-2.5 h-4 sm:h-4 lg:h-5 bg-primary dark:bg-dark-primary inline-block ml-1 "
                />
              </p>
            </div>
            <div
              class="flex place-items-center gap-[30px]"
            >
              <a
                class="hover:underline flex place-items-center gap-[15px]"
                href="https://www.linkedin.com/in/antoniozanotti/"
                target="_blank"
                title="Linked In"
              >
                <svg
                  class="w-[38px] h-[38px] lg:w-[48px] lg:h-[48px]"
                  fill="none"
                  role="img"
                  viewBox="0 0 48 48"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    Linked In Brand
                  </title>
                  <g
                    clip-path="url(#clip0_44_42)"
                    filter="url(#filter0_i_44_42)"
                  >
                    <path
                      d="M0 3.438C0 1.539 1.578 0 3.525 0H44.475C46.422 0 48 1.539 48 3.438V44.562C48 46.461 46.422 48 44.475 48H3.525C1.578 48 0 46.461 0 44.562V3.438ZM14.829 40.182V18.507H7.626V40.182H14.829ZM11.229 15.546C13.74 15.546 15.303 13.884 15.303 11.802C15.258 9.675 13.743 8.058 11.277 8.058C8.811 8.058 7.2 9.678 7.2 11.802C7.2 13.884 8.763 15.546 11.181 15.546H11.229ZM25.953 40.182V28.077C25.953 27.429 26.001 26.781 26.193 26.319C26.712 25.026 27.897 23.685 29.889 23.685C32.496 23.685 33.537 25.671 33.537 28.587V40.182H40.74V27.75C40.74 21.09 37.188 17.994 32.448 17.994C28.626 17.994 26.913 20.094 25.953 21.573V21.648H25.905C25.9209 21.623 25.9369 21.598 25.953 21.573V18.507H18.753C18.843 20.541 18.753 40.182 18.753 40.182H25.953Z"
                      fill="currentColor"
                    />
                  </g>
                  <defs>
                    <filter
                      color-interpolation-filters="sRGB"
                      filterUnits="userSpaceOnUse"
                      height="52"
                      id="filter0_i_44_42"
                      width="48"
                      x="0"
                      y="0"
                    >
                      <feflood
                        flood-opacity="0"
                        result="BackgroundImageFix"
                      />
                      <feblend
                        in="SourceGraphic"
                        in2="BackgroundImageFix"
                        mode="normal"
                        result="shape"
                      />
                      <fecolormatrix
                        in="SourceAlpha"
                        result="hardAlpha"
                        type="matrix"
                        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                      />
                      <feoffset
                        dy="4"
                      />
                      <fegaussianblur
                        stdDeviation="2"
                      />
                      <fecomposite
                        in2="hardAlpha"
                        k2="-1"
                        k3="1"
                        operator="arithmetic"
                      />
                      <fecolormatrix
                        type="matrix"
                        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                      />
                      <feblend
                        in2="shape"
                        mode="normal"
                        result="effect1_innerShadow_44_42"
                      />
                    </filter>
                    <clippath
                      id="clip0_44_42"
                    >
                      <rect
                        fill="white"
                        height="48"
                        width="48"
                      />
                    </clippath>
                  </defs>
                </svg>
                Linked In
              </a>
              <a
                class="hover:underline flex place-items-center gap-[15px]"
                href="https://github.com/antoniozanotti"
                target="_blank"
                title="GitHub"
              >
                <svg
                  class="w-[38px] h-[38px] lg:w-[48px] lg:h-[48px]"
                  fill="none"
                  role="img"
                  viewBox="0 0 48 48"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    GitHub Brand
                  </title>
                  <g
                    filter="url(#filter0_i_44_44)"
                  >
                    <path
                      d="M24 0C10.74 0 0 10.74 0 24C0 34.62 6.87 43.59 16.41 46.77C17.61 46.98 18.06 46.26 18.06 45.63C18.06 45.06 18.03 43.17 18.03 41.16C12 42.27 10.44 39.69 9.96 38.34C9.69 37.65 8.52 35.52 7.5 34.95C6.66 34.5 5.46 33.39 7.47 33.36C9.36 33.33 10.71 35.1 11.16 35.82C13.32 39.45 16.77 38.43 18.15 37.8C18.36 36.24 18.99 35.19 19.68 34.59C14.34 33.99 8.76 31.92 8.76 22.74C8.76 20.13 9.69 17.97 11.22 16.29C10.98 15.69 10.14 13.23 11.46 9.93C11.46 9.93 13.47 9.3 18.06 12.39C19.98 11.85 22.02 11.58 24.06 11.58C26.1 11.58 28.14 11.85 30.06 12.39C34.65 9.27 36.66 9.93 36.66 9.93C37.98 13.23 37.14 15.69 36.9 16.29C38.43 17.97 39.36 20.1 39.36 22.74C39.36 31.95 33.75 33.99 28.41 34.59C29.28 35.34 30.03 36.78 30.03 39.03C30.03 42.24 30 44.82 30 45.63C30 46.26 30.45 47.01 31.65 46.77C36.4147 45.162 40.555 42.1 43.4881 38.0153C46.4211 33.9305 47.9992 29.0287 48 24C48 10.74 37.26 0 24 0Z"
                      fill="currentColor"
                    />
                  </g>
                  <defs>
                    <filter
                      color-interpolation-filters="sRGB"
                      filterUnits="userSpaceOnUse"
                      height="52"
                      id="filter0_i_44_44"
                      width="48"
                      x="0"
                      y="0"
                    >
                      <feflood
                        flood-opacity="0"
                        result="BackgroundImageFix"
                      />
                      <feblend
                        in="SourceGraphic"
                        in2="BackgroundImageFix"
                        mode="normal"
                        result="shape"
                      />
                      <fecolormatrix
                        in="SourceAlpha"
                        result="hardAlpha"
                        type="matrix"
                        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                      />
                      <feoffset
                        dy="4"
                      />
                      <fegaussianblur
                        stdDeviation="2"
                      />
                      <fecomposite
                        in2="hardAlpha"
                        k2="-1"
                        k3="1"
                        operator="arithmetic"
                      />
                      <fecolormatrix
                        type="matrix"
                        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.25 0"
                      />
                      <feblend
                        in2="shape"
                        mode="normal"
                        result="effect1_innerShadow_44_44"
                      />
                    </filter>
                  </defs>
                </svg>
                GitHub
              </a>
            </div>
            <div
              class="pb-4 sm:pb-8 lg:pb-16 flex place-content-center"
            >
              <button
                aria-label="Go to About Section"
                class="rounded-full font-medium flex gap-x-2 justify-center place-items-center focus:outline focus:outline-offset-2 focus:outline-2  bg-secondary dark:bg-dark-secondary hover:bg-secondary-1 active:bg-secondary-2 focus:outline-secondary dark:hover:bg-dark-secondary-1 dark:active:bg-dark-secondary-2 dark:focus:outline-dark-secondary text-inverse-secondary dark:text-inverse-dark-secondary h-[38px] lg:h-[48px] min-w-[38px] lg:min-w-[48px] "
              >
                <svg
                  aria-hidden="true"
                  class="w-[16px] h-[16px] lg:w-[20px] lg:h-[20px] text-inverse-secondary dark:text-inverse-dark-secondary "
                  data-slot="icon"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title>
                    ArrowDownIcon
                  </title>
                  <path
                    d="M19.5 13.5 12 21m0 0-7.5-7.5M12 21V3"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </button>
            </div>
          </div>
        </header>
      </div>
    </div>
  </div>
</div>
`;
